### DISCUSSION TITLE
Will there be a web support?

### DISCUSSION THREAD

**Discussion by engblnk** [ra1u/redis-dart#27] [Ideas]
Thanks for the good work.

**Comment by ra1u** [comment]
Hi @engblnk 

Can you please, better explain what you have in mind by web support?

**Comment by engblnk** [comment]
Right :),
I intend to build a dynamic webpage, basicly updating lists of data streaming from redis.
I want use flutter web since I have already some experience in flutter.

In https://pub.dev/packages/redis it states that, this package supports iOS and Android but Web is not there.

Best Regards,
Engin

**Comment by ra1u** [comment]
I will take a look into what can be done. 

Current implementation is using raw socket to connect on redis, where web browsers can communicate only trough http. 

@engblnk  how do you plan to connect on redis?

**Comment by engblnk** [comment]
I get it now. I am not sure.

The lists will be highly volatile, up to 5000 updates in every second. 
@ra1u Should I consider any performance issues if I use http?

**Comment by ra1u** [comment]
Handling lot of  http request concurrently in web-browser, particularly in java script might be a tricky.  However if your concern is performance and you are unable to aggregate requests you can try using websockets. 

For this you would need to get websocket translator to redis (i am not sure that redis provides websocket connection), and modify redis-dart library for support of web-sockets instead of just sockets.

**Comment by engblnk** [comment]
I understand now, thank you very much.
Now, things got complicated then it suppose to be. :)

**Comment by abhin-vicks** [comment]
Please let us know if any update comes for web support.

**Comment by TheOneWithTheBraid** [comment]
I think it will be nether possible nor useful to have redis in web. In such a use case, a server side implementation featuring websocket communication might be more useful.

**Comment by etozzato** [comment]
I think I found a use case that would benefit from web support: I am working on a POC `deskop` app where user authentication and role management is backed by Redis on the local network. Instead of using REST against a backed that uses Redis like we we normally do, the app is creating and maintaining user sessions using Redis directly. 

I will be looking into `web_socket_channel` to see if it's even possible to send commands to Redis using websockets!

**Reply by ra1u** [reply]
Basic problem for JS support is that redis as far as I know supports only raw tcpip connection,  but JS from browser does not allow raw tcp, but websocket is best alternative. To have Js support  we would need dart package, that does not use raw tcp/ip as currently, but only  [WebSocket-class](https://api.dart.dev/stable/2.14.4/dart-html/WebSocket-class.html). 
Second package compatible with current interface/library on top of this WebSocket-dart package would be able to replace websocket interface with raw io-interface.

For websockets, user will still need to run additional reverse-proxy server on side with redis server,  that would translate web-socket connection to raw tpcip stream.

**Reply by etozzato** [reply]
Something like this? Overcomplicated, but the idea of accessing Redis from the browser is appealing. 

```
([Flutter][JS][ws-client]) <---> ([ws-server][socket]) <---> ([socket][redis])
```

**Reply by ra1u** [reply]
That is correct. I am not sure how popular this approach is. Redis is meant to be run on lan network  due to low latency and tight security. Exposing Redis over internet brings security and performance issues. That is - if you are executing redis commands over internet on Redis database, then you are doing something wrong and your system design needs reconsideration.
